proposals/1113_ldap-ad-web-ui/README.md

```diff
@@ -0,0 +1,151 @@
+---
+title: "Web UI for LDAP/AD login"
+authors: [ "@margocrawf" ]
+status: "draft"
+approval_date: ""
+---
+
+*Disclaimer*: Proposals are point-in-time designs and decisions.
+Once approved and implemented, they become historical documents.
+If you are reading an old proposal, please be aware that the
+features described herein might have continued to evolve since.
+
+# Web UI for LDAP/AD login
+
+## Problem Statement
+Today the supervisor only supports a single, hard coded public OAuth client called
+"pinniped-cli" which supports the pinniped CLI’s interactions with the Pinniped Supervisor.
+When clients log in to their IDPs using LDAP or Active Directory, they are prompted to enter their
+credentials the Pinniped CLI without a browser opening.
+The pinniped cli sends the client credentials to the Supervisor, which sends them to the identity provider.
+The "pinniped-cli" client is privileged and as such is trusted to handle a user's credentials
+when authenticating with systems that do not provide an authentication UI (i.e. LDAP).
+
+However, Pinniped is planning to introduce support for dynamic OAuth clients.
+These clients should _not_ be trusted to handle a user's IDP credentials.
+Therefore, we need a mechanism for untrusted clients to acquire Pinniped's downstream tokens while
+leaving the IDP credential handling to the Pinniped supervisor.
+
+## Proposal
+Pinniped must provide a simple login screen in order to support UIs that wish
+to authenticate with the Pinniped Supervisor to gain access to a cluster without
+requiring each app to handle IDP credentials.
+
+### Goals and Non-goals
+
+#### Goals
+* Prevent OAuth clients, other than the Pinniped CLI, from providing credentials via the authorization request
+* Provide a minimal feature set (ie user id, password & submit button only)
+* Provide generalized error messaging for failed logins that do not expose sensitive information (i.e. we should say "invalid username or password"
+  but do not expose whether it's the username or password that's incorrect)
+* Provide information easily allowing a user to identify the screen as belonging to Pinniped and which upstream IdP is being represented (e.g. IdP name)
+* Address basic security concerns for web firms (HTTPS, passwords use a password field, CSRF protection, redirect protection)
+* Prevent LDAP injection attacks
+* Rely on the upstream IdP to address advanced security concerns (brute force protection, username enumeration, etc)
+* Screens are accessible and friendly to screen readers
+* Screens are friendly to password managers
+
+#### Non-goals
+* A rich client (ie the use of javascript)
+* Advanced UI features (e.g. remember me, reveal password). These features are better left to identity providers to implement.
```

Comment:
Small nit: We can't leave these particular features to the actual identity provider to implement, since this page will be inside the Supervisor's code. However, the point about these being non-goals for the first version of this page still stands.